src/typeChecker/__tests__/source4TypedModules.test.ts

```diff
@@ -0,0 +1,287 @@
+import { mockContext } from '../../mocks/context'
+import { Chapter, Variant } from '../../types'
+import { parse } from '../../parser/parser'
+import { parseError } from '../../index'
+
+let context = mockContext(Chapter.SOURCE_4, Variant.TYPED)
+
+beforeEach(() => {
+  context = mockContext(Chapter.SOURCE_4, Variant.TYPED)
+  context.nativeStorage.loadedModules = {
+    module: {}
+  }
+  context.nativeStorage.loadedModulesTypes = {
+    module: {
+      prelude: `
+        class Test1 {}
+        class Test2 {}
+        class Test3 {}
+      `,
+      x: 'const x: string = "hello"',
+      y: 'const y: number = 42',
+      z: 'const z: boolean = true',
+      test1: 'const test1: Test1 = Test1',
+      test2: 'const test2: Test2 = Test2',
+      test3: 'const test3: Test3 = Test3',
+      function1: 'function function1(n: number, s: string, b: boolean): number { return n }',
+      function2: 'function function2(s: string, n: number, b: boolean): string { return s }',
+      function3: 'function function3(b: boolean, s: string, n: number): boolean { return b }',
+      functionError: 'function functionError(n: number): string { return n }'
+    }
+  }
+})
+
+describe('Typed module tests', () => {
+  /* VALID CASES */
+
+  it('should allow correct assignment of a string variable', () => {
+    const code = `
+      import { x } from 'module';
```

Comment:
Because although you can infer it this way that `module` is the name of the module, it's not immediately obvious unless you look through the code and are familiar with both the proof-of-concept and proposal.